src/Traversals/Traversals.jl

```diff
@@ -192,30 +246,108 @@ end
 
 
 """
-    tree(g, s, alg)
+    tree(g, ss, alg)
 
-Return a directed acyclic graph based on traversal of the graph `g` starting with source vertex `s`
-using algorithm `alg`.
+Return a directed acyclic graph based on traversal of the graph `g` starting with source vertex/vertices
+`ss` using algorithm `alg`.
 """
-function tree(g::AbstractGraph, s::Integer, alg::TraversalAlgorithm)
-    p = parents(g, s, alg)
-    return tree(p)
+tree(g::AbstractGraph, ss, alg::TraversalAlgorithm) = tree(parents(g, ss, alg))
+
+"""
+    distances(s::TraversalState)
+
+Return a vector filled with the distances previously calculated and stored in
+[`TraversalState`](@ref) `s`. Unreachable vertices are indicated by a distance
+of `typemax(eltype(distances(s)))`.
+"""
+@inline distances(s::TraversalState) = s.distances
+
+mutable struct DistanceState{T<:Integer} <: TraversalState
+    distances::Vector{T}
+    n_level::T
 end
 
+@inline function initfn!(s::DistanceState{T}, u) where T 
```

Comment:
really not sure the inline should be there, given that we don't use any FFI whatsoever, it should be up to the compiler to determine whether to inline or not

Reply:
Probably not, but I wanted to be sure. If you can flag the `@inline` statements we can get rid of, I'll do it - but I'd like to be consistent - if one visitfn has no `@inline` statement, the rest should follow suit.

Reply:
Following discussion on slack, I've removed all `@inline` statements in this PR.